torch/serialization.py

```diff
@@ -121,8 +121,18 @@ def _with_file_like(f, mode, body):
             f.close()
 
 
+def _is_compressed_file(f):
+    compress_modules = ['gzip']
+    try:
+        return f.__module__ in compress_modules
```

Comment:
How hard would it be to blacklist everything that isn't gzip and the regular file object from open, so that users don't accidentally load random things?

Reply:
I'm not so sure about this. If I have a dummy file object that supports read and seek (and maybe some extra ones) I would expect it to work as a file object.

Reply:
That's a really good idea, but it might be a bit complicated. A excerpt from docs of `open()`:

> The type of file object returned by the open() function depends on the mode. When open() is used to open a file in a text mode ('w', 'r', 'wt', 'rt', etc.), it returns a subclass of io.TextIOBase (specifically io.TextIOWrapper). When used to open a file in a binary mode with buffering, the returned class is a subclass of io.BufferedIOBase. The exact class varies: in read binary mode, it returns an io.BufferedReader; in write binary and append binary modes, it returns an io.BufferedWriter, and in read/write mode, it returns an io.BufferedRandom. When buffering is disabled, the raw stream, a subclass of io.RawIOBase, io.FileIO, is returned.